Add algerian states.

senadir opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Adds Algerian states (provinces) to WooCommerce so they can be used when setting up shipping zones.

I'm tagging along a couple of Algerian community translators for a sanity check. @aladin002dz @Fcmam5 I'd appreciate a confirmation that all of those names are correct and in English (except Alger, I wasn’t sure if I should go with Alger or Algiers).
Keys use [ISO 3166-2:DZ](https://en.wikipedia.org/wiki/ISO_3166-2:DZ).
This data was populated from https://github.com/mohsenuss91/AlgerianAdministrativeDivision by @mohsenuss91.
![image](https://user-images.githubusercontent.com/6165348/74654681-287ff580-518b-11ea-8be1-a061b222c157.png)


### How to test the changes in this Pull Request:

1. Open shipping zones and try to add a region within Algeria.

### Changelog entry

> Localization - Added Algerian provinces.

senadir committed:
add algerian states

Fcmam5 commented on i18n/states.php:
```diff
@@ -292,6 +292,56 @@
+		'DZ-19' => __( 'Setif', 'woocommerce' ),
+		'DZ-20' => __( 'Saida', 'woocommerce' ),
+		'DZ-21' => __( 'Skikda', 'woocommerce' ),
+		'DZ-22' => __( 'Sidi bel abbes', 'woocommerce' ),
```
```suggestion
		'DZ-22' => __( 'Sidi Bel Abbès', 'woocommerce' ),
```

Fcmam5 commented on i18n/states.php:
```diff
@@ -292,6 +292,56 @@
+		'DZ-23' => __( 'Annaba', 'woocommerce' ),
+		'DZ-24' => __( 'Guelma', 'woocommerce' ),
+		'DZ-25' => __( 'Constantine', 'woocommerce' ),
+		'DZ-26' => __( 'Medea', 'woocommerce' ),
```
```suggestion
		'DZ-26' => __( 'Médéa', 'woocommerce' ),
```

Fcmam5 commented on i18n/states.php:
```diff
@@ -292,6 +292,56 @@
+		'DZ-25' => __( 'Constantine', 'woocommerce' ),
+		'DZ-26' => __( 'Medea', 'woocommerce' ),
+		'DZ-27' => __( 'Mostaghanem', 'woocommerce' ),
+		'DZ-28' => __( 'Msila', 'woocommerce' ),
```
```suggestion
		'DZ-28' => __( 'M\'Sila', 'woocommerce' ),
```

Fcmam5 commented on i18n/states.php:
```diff
@@ -292,6 +292,56 @@
+		'DZ-42' => __( 'Tipaza', 'woocommerce' ),
+		'DZ-43' => __( 'Mila', 'woocommerce' ),
+		'DZ-44' => __( 'Ain Defla', 'woocommerce' ),
+		'DZ-45' => __( 'Naama', 'woocommerce' ),
```
```suggestion
		'DZ-45' => __( 'Naâma', 'woocommerce' ),
```

claudiosanches replied:
@Fcmam5 thanks for your help, but I'll keep `Naama` what is the accepted value for the CLDR version 37 list, see: https://github.com/unicode-org/cldr/blob/release-37/common/subdivisions/en.xml#L1176-L1224

Fcmam5 commented on i18n/states.php:
```diff
@@ -292,6 +292,56 @@
+		'DZ-43' => __( 'Mila', 'woocommerce' ),
+		'DZ-44' => __( 'Ain Defla', 'woocommerce' ),
+		'DZ-45' => __( 'Naama', 'woocommerce' ),
+		'DZ-46' => __( 'Ain Timouchent', 'woocommerce' ),
```
```suggestion
		'DZ-46' => __( 'Aïn Témouchent', 'woocommerce' ),
```

Fcmam5 commented on i18n/states.php:
```diff
@@ -292,6 +292,56 @@
+		'DZ-16' => __( 'Alger', 'woocommerce' ),
+		'DZ-17' => __( 'Djelfa', 'woocommerce' ),
+		'DZ-18' => __( 'Jijel', 'woocommerce' ),
+		'DZ-19' => __( 'Setif', 'woocommerce' ),
```
```suggestion
		'DZ-19' => __( 'Sétif', 'woocommerce' ),
```

Fcmam5 commented on i18n/states.php:
```diff
@@ -292,6 +292,56 @@
+		'DZ-17' => __( 'Djelfa', 'woocommerce' ),
+		'DZ-18' => __( 'Jijel', 'woocommerce' ),
+		'DZ-19' => __( 'Setif', 'woocommerce' ),
+		'DZ-20' => __( 'Saida', 'woocommerce' ),
```
```suggestion
		'DZ-20' => __( 'Saïda', 'woocommerce' ),
```

Fcmam5 reviewed (changes requested):
If you want to go with French names (could be the case for you so you by choosing `Alger` over `Algiers` and for writing `Béjaïa` like this :slightly_smiling_face: ) please take a look at my suggestions below.

mohsenuss91 commented:
Great work everyone, thank you for contributing! 👏

aladin002dz commented:
does look like a good idea to go with the french version of names, they will look odd, don't they?

mohsenuss91 commented:
> doesn't look like a good idea to go with the french version of names, they will look odd, don't they?

Yup, True!

Fcmam5 commented:
> > doesn't look like a good idea to go with the french version of names, they will look odd, don't they?
> 
> Yup, True!
> But unfortunately, in Algeria we are still using french names (as official names).

I agree with you @mohsenuss91 , If we want to localize, we need to adapt to what people are used to, and what they see in their IDs and in road signs...

mohsenuss91 commented:
> > > doesn't look like a good idea to go with the french version of names, they will look odd, don't they?
> > 
> > 
> > Yup, True!
> > But unfortunately, in Algeria we are still using french names (as official names).
> 
> I agree with you @mohsenuss91 , If we want to localize, we need to adapt to what people are used to, and what they see in their IDs and in road signs...

Another option is to use both (English & French) with Arabic (official name).

senadir commented:
Other translations will come by time, this initial version is the default one and is intended to be in English, the confusing was in Alger and Algiers mostly, but I think we can get that solved and merge this to get in Woo 4.1, and possibly contribute to the French and Arabic translations later.
So @Fcmam5 @mohsenuss91 what is yet needed for this to be a pure english translation? only Bejaia and Alger?

Fcmam5 commented:
> Other translations will come by time, this initial version is the default one and is intended to be in English, the confusing was in Alger and Algiers mostly, but I think we can get that solved and merge this to get in Woo 4.1, and possibly contribute to the French and Arabic translations later.
> So @Fcmam5 @mohsenuss91 what is yet needed for this to be a pure english translation? only Bejaia and Alger?

Yes, and just use caps for `Sidi Bel Abbes` instead of `Sidi bel abbes` :yum:

claudiosanches committed:
Updated algerian states based on the CLDR list

Based on https://github.com/unicode-org/cldr/blob/release-37/common/subdivisions/en.xml#L1176-L1224

claudiosanches reviewed (approved):
Looks great!

I updated the entries to match with the CLDR list: https://github.com/unicode-org/cldr/blob/release-37/common/subdivisions/en.xml#L1176-L1224
